csunplugged/utils/TextBoxDrawer.py

```diff
@@ -0,0 +1,411 @@
+"""Module for TextBoxDrawer class."""
+
+from django.utils.translation import get_language, get_language_bidi
+from PIL import ImageFont, Image, ImageDraw
+from lxml import etree as ET
+import tinycss
+import os
+from bidi.algorithm import get_display
+from uniseg.linebreak import line_break_units
+import arabic_reshaper
+import math
+from utils.errors.TextBoxDrawerErrors import (
+    MissingSVGFile,
+    TextBoxNotFoundInSVG
+)
+
+DEFAULT_FONT = "static/fonts/NotoSans-Regular.ttf"
+DEFAULT_FONT_OVERRIDES = {
+    "ja": "static/fonts/NotoSansCJKjp-Regular.otf",
+    "he": "static/fonts/OpenSansHebrew-Regular.ttf",
+    "ar": "static/fonts/DejaVuSans.ttf",
+}
+DEFAULT_FONT_SIZE = 20
+DEFAULT_COLOR = (0, 0, 0)
+
+FONT_MAX_ORD = {
```

Comment:
`ord` = Ordinal of the character, or its unicode code point. In python, the function `ord` is used - i.e.
```
>>> ord("a")
97
>>> ord("は")
12399
```

Many fonts only contain glyphs for a small subset of unicode - in this case, I believed PatrickHand-Regular only contains glyphs for characters up to 0xFF (255). This dictionary is used to map fonts to the highest ordinal for which they contain a glyph. For strings containing chars with a higher ordinal than this, a fallback font will have to be used.

On further research, it seems that PatrickHand-Regular actually supports more codepoints than I first thought. Ideally, we would extract this information directly from the font file rather than hardcoding it in, although a bit of quick research made this seem non-trivial. It may be possible using the `fontTools` library, I will investigate tomorrow.

Reply:
It looks like an approach similar to https://stackoverflow.com/questions/43834362 may work.

Reply:
That solution looks smarter, but also Patrick Hand will be gutted from the system soon and replaced by Noto.

Reply:
As discussed offline, this will be left as is for now. It will be improved in a later PR (tracked in issue #671)